doc/user_guide/psyir.rst

```diff
@@ -445,12 +446,53 @@ in some cases children must be collected together within a
 To simplify this complexity, each of the Kernel-layer nodes which
 contain other nodes have a static ``create`` method which helps
 construct the PSyIR using a bottom up approach. Using this method, the
-above example then becomes::
+above example then becomes:
+
+.. code-block:: python
 
-    > ...
-    > literal = Literal("0.0", REAL_TYPE)
-    > reference = Reference(symbol)
-    > assignment = Assignment.create(reference, literal)
+    literal = Literal("0.0", REAL_TYPE)
+    reference = Reference(symbol)
+    assignment = Assignment.create(reference, literal)
 
 More complete examples of using this approach can be found in the
 PSyclone ``examples/psyir`` directory.
+
+
+Modifying the PSyIR
+===================
+
+Once we have a complete PSyIR AST created, the PSyIR also provides methods to
+modify the tree in a consistent way (e.g. without breaking its many internal
+references). Some complete examples of modifying the PSyIR can be found in the
+PSyclone ``examples/psyir/modify.py`` script.
+Below there are examples of possible PSyIR modifications:
+
+Renaming symbols
+-----------------
+The symbol table provides the method ``rename_symbol()`` that given a symbol
+and a unused name it will rename the symbol an affect all the references in
+the PSyIR tree to that symbol. For example, the PSyIR representing the
+following Fortran code:
+
+.. code-block:: fortran
+
+    subroutine work(psyir_tmp)
+        real, intent(inout) :: psyir_tmp
+        psyir_tmp=0.0
+    end subroutine
+
+Could be modified by PSyIR statements:
+
+.. code-block:: python
+
+    symbol = symbol_table.lookup("psyir_tmp")
+    symbol_table.rename_symbol(tmp_symbol, "new_variable")
+
+To obtain the representation for:
```

Comment:
which would result in the following Fortran output code:

Reply:
Done